go/vt/vtadmin/api.go

```diff
@@ -534,6 +540,47 @@ func (api *API) getTablets(ctx context.Context, c *cluster.Cluster) ([]*vtadminp
 	return ParseTablets(rows, c)
 }
 
+// findTablet returns the first tablet in a given cluster that satisfies the filter function.
```

Comment:
I would definitely like to move them elsewhere. I think, though, that moving them to `cluster.Cluster` might introduce a weird dependency. 

Specifically, per your suggestion, I want to move the following three functions out of [vtadmin/api.go](https://github.com/vitessio/vitess/blob/a93827729719f83c0bd8e1becea7e05de8398e7a/go/vt/vtadmin/api.go) and into [vtadmin/cluster/cluster.go](https://github.com/vitessio/vitess/blob/a93827729719f83c0bd8e1becea7e05de8398e7a/go/vt/vtadmin/cluster/cluster.go):

- [findTablet](https://github.com/vitessio/vitess/blob/a93827729719f83c0bd8e1becea7e05de8398e7a/go/vt/vtadmin/api.go#L544)
- [findTablets](https://github.com/vitessio/vitess/blob/a93827729719f83c0bd8e1becea7e05de8398e7a/go/vt/vtadmin/api.go#L557)
- [getTablets](https://github.com/vitessio/vitess/blob/a93827729719f83c0bd8e1becea7e05de8398e7a/go/vt/vtadmin/api.go#L530), which is called by the two above

However, `getTablets` calls `ParseTablets, which is in [vtadmin/tablets.go](https://github.com/vitessio/vitess/blob/a93827729719f83c0bd8e1becea7e05de8398e7a/go/vt/vtadmin/tablets.go) (`package "vtadmin"), which would (I think) cause a circular dependency. I assume this is undesirable in golang, right? 

I have two ideas, not sure which (if either) make sense to do: 
- Create a separate `package tablet` `vtadmin/tablet/tablet.go` (analogous to `vtadmin/cluster/cluster.go`)
- Move the three functions listed above into `vtadmin/tablets.go`, since the _other_ functions in `tablets.go` also take a cluster argument. It seems like not a bad fit. 

Happy to take any suggestion you have!

Reply:
Yeah, I think anything that is like `func (api *API) DoThingWithTablets(ctx context.Context, c *cluster.Cluster)` we should _try_ (it might have some other dependencies on `api`) to make it a `func (c *Cluster) DoThingWithTablets(ctx context.Context)`

Reply:
@ajm188 hmmm ok. In that case, I _will_ have to move `tablet.go` (and its `ParseTablet` function) into its own `package tablet` so `cluster.go` can use it, right? It seems reasonable to me, I just want to double check before I move things around.

Reply:
Yeah, that's what I was saying with where it should be `func (c *Cluster) ParseTablets(ctx context.Context, ...) ([]*vtadminpb.Tablet, error)` instead of `func (api *API) ParseTablets(ctx context.Context, c *cluster.Cluster, ...) (...)`.

Also, I'm not super convinced that needs to remain a public function. It doesn't really feel like part of the API of a cluster, while "find N tablets in the cluster" definitely does.

Reply:
@ajm188 that sounds really good. 

Moving things around ends up being a rather big + messy commit, and it'd be nice to merge this one in since I'm branched off of it for front-end reasons. 

Would it be ok with you if I moved everything into `clusters.go` in a separate PR? I can do it today, right after this, and add a bunch of tests as penance.

Reply:
Yeah, let's file an issue for that, because I'll forget, but sg2m to merge